src/mame/machine/nes_vt369_vtunknown_soc.cpp

```diff
@@ -0,0 +1,274 @@
+{
+}
+
+nes_vtunknown_soc_dg_device::nes_vtunknown_soc_dg_device(const machine_config& mconfig, const char* tag, device_t* owner, uint32_t clock) :
+	nes_vtunknown_soc_dg_device(mconfig, NES_VTUNKNOWN_SOC_DG, tag, owner, clock)
+{
+}
+
+nes_vtunknown_soc_fa_device::nes_vtunknown_soc_fa_device(const machine_config& mconfig, const char* tag, device_t* owner, uint32_t clock) :
+	nes_vtunknown_soc_dg_device(mconfig, NES_VTUNKNOWN_SOC_FA, tag, owner, clock)
+{
+}
+
+/***********************************************************************************************************************************************************/
+/* 'CY' specifics (base = '4K') */
+/***********************************************************************************************************************************************************/
+
+void nes_vtunknown_soc_cy_device::device_add_mconfig(machine_config& config)
+{
+	nes_vt02_vt03_soc_device::device_add_mconfig(config);
+	m_maincpu->set_addrmap(AS_PROGRAM, &nes_vtunknown_soc_cy_device::nes_vt_cy_map);
+}
+
+void nes_vtunknown_soc_cy_device::nes_vt_cy_map(address_map &map)
+{
+	nes_vt_4k_ram_map(map);
+	map(0x41b0, 0x41bf).r(FUNC(nes_vtunknown_soc_cy_device::vt03_41bx_r)).w(FUNC(nes_vtunknown_soc_cy_device::vt03_41bx_w));
+	map(0x4130, 0x4136).r(FUNC(nes_vtunknown_soc_cy_device::vt03_413x_r)).w(FUNC(nes_vtunknown_soc_cy_device::vt03_413x_w));
+	map(0x414f, 0x414f).r(FUNC(nes_vtunknown_soc_cy_device::vt03_414f_r));
+	map(0x415c, 0x415c).r(FUNC(nes_vtunknown_soc_cy_device::vt03_415c_r));
+
+	map(0x48a0, 0x48af).r(FUNC(nes_vtunknown_soc_cy_device::vt03_48ax_r)).w(FUNC(nes_vtunknown_soc_cy_device::vt03_48ax_w));
+}
+
+void nes_vtunknown_soc_cy_device::device_start()
+{
+	nes_vt02_vt03_soc_device::device_start();
+	save_item(NAME(m_413x));
+}
+
+
+uint8_t nes_vtunknown_soc_cy_device::vt03_41bx_r(offs_t offset)
+{
+	switch (offset)
+	{
+	case 0x07:
+		return 0x04;
+	default:
+		return 0x00;
+	}
+}
+
+void nes_vtunknown_soc_cy_device::vt03_41bx_w(offs_t offset, uint8_t data)
+{
+	logerror("vt03_41bx_w %02x %02x\n", offset, data);
+}
+
+uint8_t nes_vtunknown_soc_cy_device::vt03_413x_r(offs_t offset)
+{
+	logerror("vt03_413x_r %02x\n", offset);
+	return m_413x[offset];
+}
+
+void nes_vtunknown_soc_cy_device::vt03_413x_w(offs_t offset, uint8_t data)
+{
+	logerror("vt03_413x_w %02x %02x\n", offset, data);
+	// VT168 style ALU ??
+	m_413x[offset] = data;
+	if (offset == 0x5)
+	{
+		uint32_t res = uint32_t((m_413x[5] << 8) | m_413x[4]) * uint32_t((m_413x[1] << 8) | m_413x[0]);
+		m_413x[0] = res & 0xFF;
+		m_413x[1] = (res >> 8) & 0xFF;
+		m_413x[2] = (res >> 16) & 0xFF;
+		m_413x[3] = (res >> 24) & 0xFF;
+		m_413x[6] = 0x00;
+
+	}
+	else if (offset == 0x6)
+	{
+		/*uint32_t res = uint32_t((m_413x[5] << 8) | m_413x[4]) * uint32_t((m_413x[1] << 8) | m_413x[0]);
+		m_413x[0] = res & 0xFF;
+		m_413x[1] = (res >> 8) & 0xFF;
+		m_413x[2] = (res >> 16) & 0xFF;
+		m_413x[3] = (res >> 24) & 0xFF;*/
+		m_413x[6] = 0x00;
+	}
```

Comment:
Why uppercase hex here when you’ve used lowercase throughout the rest of the file?  Please be consistent.

Reply:
likewise, this isn't new code, it's moved code.